collated/main/chrisboo.md

```diff
@@ -0,0 +1,434 @@
+    }
+
+    @Override
+    public CommandResult execute() {
+        model.updateFilteredPersonList(person -> findPersonDescriptor.match(person));
+        return new CommandResult(getMessageForPersonListShownSummary(model.getFilteredPersonList().size()));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof FindCommand // instanceof handles nulls
+            && this.findPersonDescriptor.equals(((FindCommand) other).findPersonDescriptor)); // state check
+    }
+
+    /**
+     * Stores the details to find of the person.
+     */
+    public static class FindPersonDescriptor {
+        private Name name;
+        private Phone phone;
+        private Email email;
+        private Address address;
+        private Birthday birthday;
+        private Website website;
+        private Set<Tag> tags;
+
+        public FindPersonDescriptor() {
+        }
+
+        public FindPersonDescriptor(FindPersonDescriptor toCopy) {
+            this.name = toCopy.name;
+            this.phone = toCopy.phone;
+            this.email = toCopy.email;
+            this.address = toCopy.address;
+            this.birthday = toCopy.birthday;
+            this.website = toCopy.website;
+            this.tags = toCopy.tags;
+        }
+
+        /**
+         * @return false if no fields are provided
+         */
+        public boolean allNull() {
+            return this.name == null
+                && this.phone == null
+                && this.email == null
+                && this.address == null
+                && this.birthday == null
+                && this.website == null
+                && this.tags == null;
+        }
+
+        /**
+         * @param other to check
+         * @return true if other matches all fields
+         */
+        public boolean match(Object other) {
+            if (!(other instanceof Person)) {
+                return false;
+            }
+            if (this.name != null && !match(this.name, ((Person) other).getName())) {
+                return false;
+            }
+            if (this.phone != null && !match(this.phone, ((Person) other).getPhone())) {
+                return false;
+            }
+            if (this.email != null && !this.email.equals(((Person) other).getEmail())) {
+                return false;
+            }
+            if (this.address != null && !this.address.equals(((Person) other).getAddress())) {
+                return false;
+            }
+            if (this.website != null && !this.website.equals(((Person) other).getWebsite())) {
+                return false;
+            }
+            if (this.birthday != null && !this.birthday.equals(((Person) other).getBirthday())) {
+                return false;
+            }
+            if (this.tags != null && !match(this.tags, ((Person) other).getTags())) {
+                return false;
+            }
```

Comment:
There is a code fragment being repeated. See if you can extract out the similarity.